lib/mayaUsd/render/mayaToHydra/renderOverride.cpp

```diff
@@ -377,8 +386,44 @@ MStatus MtohRenderOverride::Render(const MHWRender::MDrawContext& drawContext) {
         _taskController->SetCameraViewport(viewport);
 #endif // USD_VERSION_NUM >= 1910
 
-        auto tasks = _taskController->GetRenderingTasks();
+        HdTaskSharedPtrVector tasks = _taskController->GetRenderingTasks();
+
+        // For playblasting, a glReadPixels is going to occur sometime after we return.
+        // But if we call Execute on all of the tasks, then z-buffer fighting may occur
+        // because every colorize/present task is going to be drawing a full-screen quad
+        // with 'unconverged' depth.
+        //
+        // To work arround this (for not Storm) we pull the first task, (render/synch)
+        // ad continually execute it until the renderer signals converged, at which point
+        // we break and call HdEngine::Execute once more to copy the aovs into OpenGL
+        //
+        if (_playBlasting && !_isUsingHdSt && !tasks.empty()) {
+            // XXX: Is this better as user-configurable ?
+            constexpr auto msWait = std::chrono::duration<float, std::milli>(100);
+
+            std::shared_ptr<HdxProgressiveTask> progressiveTask =
+                std::dynamic_pointer_cast<HdxProgressiveTask>(tasks.front());
+            if (progressiveTask) {
+                HdTaskSharedPtrVector renderOnly = { progressiveTask };
+                _engine.Execute(_renderIndex, &renderOnly);
+
+                while (!progressiveTask->IsConverged()) {
```

Comment:
Is there an easy way that we can add a check to see if the user has requested a cancel to this loop?  Even in normal situations, convergence could take a while, and in worst case, the underlying renderer has some sort of a bug which makes it never properly converge.

Reply:
Uhhh yeah, I seem to have dropped the **_playBlasting** check in a refactor..

Reply:
Awesome, thanks for the fix!